src/app/reset-password/reset-password.component.html

```diff
@@ -0,0 +1,55 @@
+<div class="b-container">
+    <biggive-heading
+      space-above=0
+      space-below=3
+      colour="black"
+      html-element="h1"
+      size="1"
+      align="left"
+      text="Reset your Password"
+    ></biggive-heading>
+    <!-- Note: check explicitly for `undefined` here, not for `!saveSuccessful`, because
+    `saveSuccessful` has can be undefined, false, or true, all with a unique meaning. -->
+    <div *ngIf="saveSuccessful === undefined && !savingNewPassword">
+      <mat-dialog-content class="mat-typography">
+        <form [formGroup]="passwordForm">
+          <p class="b-rt-0">Please enter your new password (min 10 characters)</p>
+          <mat-form-field color="primary">
+              <mat-label>Password</mat-label>
+              <input matInput type="password" formControlName="password">
+          </mat-form-field>
+          <mat-form-field color="primary">
+              <mat-label>Re-enter your password</mat-label>
+              <input matInput type="password" formControlName="confirmPassword" (focus)="onPasswordConfirmationFocus()">
+          </mat-form-field>
+          <div *ngIf="confirmPasswordField?.errors?.notSame" class="error">Passwords must be the same!</div>
+        </form>
+      </mat-dialog-content>
+      
+      <mat-dialog-actions>
+          <button
+            id="reset-modal-submit"
+            mat-raised-button
+            [disabled]="!passwordForm.valid || savingNewPassword"
+            (click)="saveNewPassword()"
+            color="primary"
+          >Save new password</button>
+      </mat-dialog-actions>
+    </div>
+
+    <!-- If processing, show spinner -->
+    <div *ngIf="saveSuccessful === undefined && savingNewPassword">
+      <mat-spinner [diameter]="40" aria-label="Loading your details"></mat-spinner>
+    </div>
+
+    <p *ngIf="saveSuccessful">
+      You have successfully reset your password. Please click <a href="/">here</a> for the homepage, or <a href="/buy-credits">here</a> to buy donation credits.
```

Comment:
Not a blocker, but the word 'successfully' is a bit of a pet peeve for me - it doesn't seem to add any useful information, and just suggest we weren't confident in things working. Could we leave it out?

```suggestion
      You have reset your password. Please click <a href="/">here</a> for the homepage, or <a href="/buy-credits">here</a> to buy donation credits.
```

Reply:
Updated